app/main.py

```diff
@@ -1,6 +1,16 @@
-from typing import Callable
+from typing import Callable, Any
 
 
-def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+def cache(func: Callable) -> Any:
+    results = {}
+
+    def wrapper(*args, **kwargs) -> Any:
+        if args in results:
```

Comment:
You should create a key based on both keyword and positional arguments, rather than using only `args` as the key. This is because the function might accept the same `args` but different `kwargs`.